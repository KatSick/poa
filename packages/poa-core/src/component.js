// @ts-check

import { observer } from 'mobx-react';
import { addComponentToRegistry } from './repository';

export function Component(config) {
  return function PoaComponent(constructor) {
    addComponentToRegistry(constructor);

    if (config && config.namespaces) {
      constructor.prototype.tNamespaces = config.namespaces;
    }

    return observer(constructor);
  };
}
