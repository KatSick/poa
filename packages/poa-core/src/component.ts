import { observer } from 'mobx-react';
import { addComponentToRegistry } from './repository';
import { PoaComponentConfig } from './interfaces/component-config.interface';

export function Component(config: PoaComponentConfig) {
  return function PoaComponent(constructor: any) {
    addComponentToRegistry(constructor);

    if (config && config.namespaces) {
      constructor.prototype.tNamespaces = config.namespaces;
    }

    return observer(constructor);
  };
}
