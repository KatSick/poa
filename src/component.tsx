import { observer } from 'mobx-react';
import { addComponentToRegistry } from './components-registry';

export interface PoaComponentDecorator {
  namespaces?: string[];
}

export function Component(config?: PoaComponentDecorator) {
  // tslint:disable-next-line:no-any
  return function PoaComponent(constructor: any) {
    addComponentToRegistry(constructor);

    if (config && config.namespaces) {
      constructor.prototype.tNamespaces = config.namespaces;
    }

    // tslint:disable-next-line:no-any
    return observer(constructor) as any;
  };
}
