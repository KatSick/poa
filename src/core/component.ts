import { observer, IReactComponent } from '../state/mobx-react';
import { ComponentsInjector } from './repository';
import { PoaComponentConfig } from './interfaces/component-config.interface';

export function PoaComponent(config?: PoaComponentConfig) {
  return function InternalPoaComponent<T extends IReactComponent>(component: T): T {
    ComponentsInjector.addComponentToRegistry(component);

    if (config && config.namespaces) {
      component.prototype.tNamespaces = config.namespaces;
    }

    return observer(component);
  };
}
