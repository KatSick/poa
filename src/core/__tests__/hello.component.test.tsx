import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { HelloComponent } from '../hello.component';

describe('CORE â€” Hello Component', () => {
  it('renders', () => {
    const component = renderer.create(<HelloComponent />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
