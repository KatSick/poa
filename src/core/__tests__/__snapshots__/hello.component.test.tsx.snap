// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CORE â€” Hello Component renders 1`] = `
<div>
  Jambo, jambo! Define your routes in 
  boot({ routes: &lt;here&gt; })
  .
</div>
`;
